server/src/main/java/org/opensearch/common/geo/GeoShapeUtils.java

```diff
@@ -62,7 +63,9 @@
     }
 
     public static org.apache.lucene.geo.Point toLucenePoint(Point point) {
-        return new org.apache.lucene.geo.Point(point.getLat(), point.getLon());
+        final double lat = GeoEncodingUtils.decodeLatitude(GeoEncodingUtils.encodeLatitude(point.getLat()));
```

Comment:
Hmm why was this round trip encode/decode needed?

Reply:
It's not... it's adding extra superfluous work! Reverting.. thx for the catch!

Reply:
Looks like a test failure as a result of reverting: 

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.geo.GeoShapeIntegrationIT.testIndexPolygonDateLine" -Dtests.seed=364ABAA5289793FA -Dtests.security.manager=true -Dtests.locale=de -Dtests.timezone=Africa/Ceuta -Druntime.java=15

org.opensearch.search.geo.GeoShapeIntegrationIT > testIndexPolygonDateLine FAILED
    java.lang.AssertionError: 
    Expected: <1L>
         but: was <0L>
        at __randomizedtesting.SeedInfo.seed([364ABAA5289793FA:DF31BD88058CBB72]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.opensearch.search.geo.GeoShapeIntegrationIT.testIndexPolygonDateLine(GeoShapeIntegrationIT.java:300)
```

Something changed in upstream Lucene land that now requires quantization. I think we'll want to explore further. In the meantime I'll add a fix that doesn't require quantizing before indexing.

Reply:
Rather than put the fix in the utility API I added it directly to the VectorGeoShapeQueryProcessor. This way quantization only happens at query time. I also added a TODO to fix this sandy Lucene experience upstream